data/hai/hai reveal 4 middle.txt

```diff
@@ -0,0 +1,463 @@
+# Copyright (c) 2020 Nomadic Volcano
+#
+# Endless Sky is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later version.
+#
+# Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+
+########################################################################
+
+# Part 4: Search Korath and human space for the missing researchers
+
+mission "Hai Rescue [A01] Visit Darkwaste"
+	landing
+	name "Missing Researchers"
+	description "Fly to <destination> to visit the Hai you rescued."
+	source
+		government "Republic" "Free Worlds" "Syndicate" "Neutral" "Independent" "Hai"
+		not attributes "uninhabited"
+		not planet "Darkwaste"
+	destination Darkwaste
+	to offer
+		has "event: hai rescue: teeneep message"
+		has "Hai Building Mountaintop [3] Furnishings: done"
+		or
+			has "Pirate Troubles [4]: done"
+			has "Pirate Troubles [4]: declined"
```

Comment:
In this case the pirate troubles set gets given an additional condition in its `to offer` which means during the process of finding and returning Hai, and then building Mountaintop you *will* trigger the pirate troubles mission somewhere.

This is because it cannot narratively happen after this, and I'm doing my best not to lock out existing content accidentally for people who haven't discovered the Hai.

But it doesn't care if you do it or decline it, just that you've had the opportunity to do either.